misc/scripts/add-repo.sh

```diff
@@ -22,47 +22,86 @@
 # You should have received a copy of the GNU General Public License
 # along with Pacstall. If not, see <https://www.gnu.org/licenses/>.
 
-REPO="${2%/}"
-
-if [[ $REPO == *"github.com"* ]]; then
-    REPO="${REPO/'github.com'/'raw.githubusercontent.com'}"
-    if [[ $REPO != *"/tree/"* ]]; then
-        REPO="$REPO/master"
-        fancy_message warn "Assuming that git branch is ${GREEN}master${NC}"
+parse_repo() {
+    local ADDR
+    IFS=':' read -ra ADDR <<< "$1"
+    PROV="${ADDR[0]}"
+    USER=$(echo "${ADDR[1]}" | cut -d'/' -f1)
+    HEAD=$(echo "${ADDR[1]}" | cut -d'/' -f2 | cut -d'#' -f1)
+    if [[ ${ADDR[1]} =~ "#" ]]; then
+        BRANCH="$(echo "${ADDR[1]}" | cut -d'#' -f2)"
     else
-        REPO="${URL/'/tree/'/'/'}"
-    fi
-elif [[ $REPO == *"gitlab.com"* ]]; then
-    if [[ $REPO != *"/tree/"* ]]; then
-        REPO="$REPO/-/raw/master"
+        BRANCH="master"
         fancy_message warn "Assuming that git branch is ${GREEN}master${NC}"
-    else
-        REPO="${REPO/"/tree/"/"/raw/"}"
     fi
-elif [[ -d $REPO ]] > /dev/null; then
-    if [[ $REPO != *"file://"* ]]; then
-        REPO="file://$(readlink -f "$REPO")"
-    fi
-else
-    fancy_message warn "The repo link must be the root to the raw files"
-    fancy_message warn "Make sure the repo contains a package list"
+}
 
-    ask "Do you want to add \"$REPO\" to the repo list?" N
-    if ((answer == 0)); then
-        exit 3
-    fi
-fi
+REPO="${2%/}"
+
+case ${REPO} in
+    *"github.com"*)
+        REPO="${REPO/'github.com'/'raw.githubusercontent.com'}"
+        if [[ $REPO != *"/tree/"* ]]; then
+            REPO="$REPO/master"
+            fancy_message warn "Assuming that git branch is ${GREEN}master${NC}"
+        else
+            REPO="${REPO/'/tree/'/'/'}"
+        fi
+        ;;
+    *"gitlab.com"*)
+        if [[ $REPO != *"/tree/"* ]]; then
+            REPO="$REPO/-/raw/master"
+            fancy_message warn "Assuming that git branch is ${GREEN}master${NC}"
+        else
+            REPO="${REPO/"/tree/"/"/raw/"}"
+        fi
+        ;;
+    *"github:"*)
+        parse_repo "${REPO}"
+        REPO="https://raw.${PROV}usercontent.com/${USER}/${HEAD}/${BRANCH}"
```

Comment:
this could potetionally be 
```bash
REPO="https://${PROV}.com/${USER}/${HEAD}/raw/${BRANCH}"
```
instead to look more equivalent to the gitlab one, but it just redirects to the usercontent link in the end anyways.

Reply:
probably not worth having mixed and matched GitHub links